### ISSUE TITLE
Complex statements...

### ISSUE THREAD

**Issue by pixlcrashr** [SkaceKamen/sqflint#6]
The last fix of complex statements just fixed the ```if``` statement.
But there are two more: the ```switch (...) do {}``` and ```try { ... } catch { ... }``` statement, which can be applied for example:

#### switch (...) do {}
```sqf
... format["%1", (
    switch (true): {
        case "example1": {};
        ...
    }
)];
...
```

#### try { ... } catch { ... }
```sqf
... format["%1", (
    try {
        ...
    } catch {
        ...
    }
)];
...
```
Because every of those clauses/statements can return only *one* type of *any* you can use them in this *commands*.
It is important to know that even the engine of BI needs some ```()``` brackets around these statements (like the array after ```format []```).

I hope you can fix this.

And for me: Happy testing :)

**Comment by SkaceKamen** [comment]
I'm aware of that, it should be ready in next version.
Updating parser so it can both find problems and allow these is a bit complex.

**Comment by LecksC** [comment]
Thanks for the great extension. The only issue we've come across (other than many 'possibly undefined vars'), is probably related to this issue, which is that if you use something like
```SQF
private _var = switch (A) do {
             case B: { X };
             case C: { Y };
};
```
It seems to throw the linter out of whack for the rest of the SQF.

These is the only false-positive marked as an 'error' in 1000+ SQFs, so great job anyway.

**Comment by SkaceKamen** [comment]
Glad it works (mostly). Do you think that the 'possible undefined' warnings are unjustified?
Assigning switch statement to variable is not yet supported, but it's close to completion, depends on my free time.

**Comment by LecksC** [comment]
For global variables i don't really see any way around the 'possible undefined' warnings unless you somehow worked out the order things are called in (not sure if there's 'information' rather than warning that would be more suitable).

Other ones are params not being interpreted properly, ie:
![image](https://cloud.githubusercontent.com/assets/13845562/24647835/dce02796-1964-11e7-8ec2-60b539720968.png)

And then there's the functions as it doesn't seem to be reading our CfgFunctions properly:
![image](https://cloud.githubusercontent.com/assets/13845562/24647706/4e73f2da-1964-11e7-8ad8-65887407f79f.png)

![image](https://cloud.githubusercontent.com/assets/13845562/24647724/66b83f04-1964-11e7-85e6-17d6efd3f297.png)
![image](https://cloud.githubusercontent.com/assets/13845562/24647730/70a51708-1964-11e7-97c9-e3bf24191390.png)

**Comment by SkaceKamen** [comment]
The params should work, I'll take look at that, thanks.

CfgFunctions parser is actually almost done (https://github.com/SkaceKamen/vscode-sqflint/issues/5), I just didn't have much time lately to fully test it and push it.

As for global variables, if you open the mission as folder, the extension should scan the folder twice, which should eliminate some of the errors.
